source/simulator/core.cc

```diff
@@ -1682,17 +1682,36 @@ namespace aspect
                                          T_preconditioner);
           solve_advection(AdvectionField::temperature());
 
+          if (parameters.use_discontinuous_temperature_discretization
+              && parameters.use_limiter_for_discontinuous_temperature_solution)
+            {
+              pcout << "   Applying bound preserving limiter to the temperature field "
+                    << std::endl;
+              apply_limiter_to_dg_solutions(AdvectionField::temperature());
+            }
+
           current_linearization_point.block(introspection.block_indices.temperature)
             = solution.block(introspection.block_indices.temperature);
 
-          for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+          for (unsigned int c=0; c < parameters.n_compositional_fields; ++c)
             {
               assemble_advection_system (AdvectionField::composition(c));
               build_advection_preconditioner(AdvectionField::composition(c),
                                              C_preconditioner);
               solve_advection(AdvectionField::composition(c));
             }
 
+          if (parameters.use_discontinuous_composition_discretization
+              && parameters.use_limiter_for_discontinuous_composition_solution)
+            {
+              pcout << "   Applying bound preserving limiter to the compositional field "
+                    << std::endl;
+              for (unsigned int c=0; c < parameters.n_compositional_fields; ++c)
+                {
+                  apply_limiter_to_dg_solutions(AdvectionField::composition(c));
+                }
```

Comment:
No need for braces around one-line statements.

In fact, my preference would be if the call to `apply_limiter...` would be placed immediately after the call to `solve_advection` above. This way, immediately after the solve/apply_limiter block, the compositional field is available in its final form.

This would lead to one line of output for each advection field. But do we actually need this line of output? I think it could just as well be deleted. The same is true for the temperature equation. Unless, of course, you really like the extra line of output :-)